### PR TITLE
e2e tests - python debugging

### LINKED ISSUE #8985: Automated tests for debugging Python notebooks (in response to #8983)
As a follow up to #8983 , I'll add automated tests for `%debug` within Python notebooks in order to verify basic functionalities and common cases. Edge cases will be tested manually herein, and will be determined if automation for such are needed.

### PR DESCRIPTION
**Pull request by rodrigosf672**
### TO DO LIST 

- [x] WIN tag
- [x] Understand and resolve issue with persisting breakpoints
- [x] Persisting breakpoints (potential reason why some tests are failing
- [x] 5-second delay for `Debug Cell`
- [x] POMerize 5-second delay (or alternative approach to hitting `Debug Cell`) -- no need to keep repeating it across tests
- [x] Inconsistent indentation pastings
- [x] Simplified, more focused test covering the basics instead of redundant, lengthy tests that are difficult to maintain in the long-term.

### QA Notes

@:web @:debug @:notebooks @:win

### CODE REVIEW

File: test/e2e/tests/notebook/notebook-debug.test.ts [hunk 1]
```diff
@@ -0,0 +1,277 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * Test Cases
+ *
+ * | Test Name                                      | Debugging Features                     | Python concept to be debuggged           | Expected Output                     |
```
[the remark this answers is not in the document]
Comment (rodrigosf672) [reply test/e2e/tests/notebook/notebook-debug.test.ts]: // small typo, in the comment 
// ignore

**Review by midleman — DISMISSED** [review]
These tests are thorough and clearly took a lot of effort... thank you for putting this together! Because e2e tests are slow and resource-intensive, I suggest we focus on one representative scenario for notebook debugging. For example, we could keep a single test that covers:

* Setting a breakpoint in a notebook cell
* Inspecting a variable at the breakpoint
* Stepping over and continuing execution
* Verifying the final output

This flow validates the core user experience end-to-end. The extra variations (arithmetic, string ops, etc.) are perfect for unit or integration tests, which are faster and easier to maintain. This way, we keep the e2e suite lean and high-value while still getting coverage where it matters. What do you think?

**Review by testlabauto — APPROVED** [review]
LGTM!
